src/commands/revealResource.ts

```diff
@@ -11,12 +11,16 @@ import { ext } from '../extensionVariables';
 import { AppResourceTreeItem } from '../tree/AppResourceTreeItem';
 import { SubscriptionTreeItem } from '../tree/SubscriptionTreeItem';
 
-export async function revealResource(context: IActionContext, resource: AppResource): Promise<void> {
-    context.telemetry.properties.resourceType = resource.type?.replace(/\//g, '|'); // Replace the slashes otherwise this gets redacted because it looks like a user file path
-    context.telemetry.properties.resourceKind = resource.kind;
+export async function revealResource(context: IActionContext, resourceId: string): Promise<void>;
+export async function revealResource(context: IActionContext, resource: AppResource): Promise<void>;
+export async function revealResource(context: IActionContext, arg: AppResource | string): Promise<void> {
```

Comment:
One thing I don't really understand is what's the point of exporting 3 different function definitions when `AppResource | string` covers the other cases? Is it just to see the arg names (`resourceId` and `resource`?)

Reply:
Yeah that way you get the two declarations, instead of one with just `AppResource | string`.